### DISCUSSION TITLE
what is the difference between res.unstable_revalidate(path) and res.revalidate(path)

### DISCUSSION THREAD

**Discussion by shmdhussain** [vercel/next.js#39908] [Help] [answered]
what is the difference between await res.unstable_revalidate(path) and res.revalidate(path) ?

In the doc mentioned as revalidate(path) https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration#on-demand-revalidation

but the blog https://nextjs.org/blog/next-12-1#on-demand-incremental-static-regeneration-beta mentioned as res.unstable_revalidate(path)

I believe unstable_revalidate(path) for early versions and revalidate(path) is for versions starting from 12.2.0.

**Comment by icyJoseph** [comment, marked as answer]
Hi,


When the feature was first released, it was behind the `unstable_revalidate, which was done in versions `12.1.x`.

Eventually the feature became stable, https://nextjs.org/blog/next-12-2#on-demand-incremental-static-regeneration-stable, and it is now called `revalidate`, but it requires versions `12.2.x`.
